booster/library/Booster/Pattern/Implies.hs

```diff
@@ -136,24 +136,31 @@ runImplies def mLlvmLibrary mSMTOptions antecedent consequent =
+                                def
+                                mLlvmLibrary
+                                solver
+                                mempty
+                                ApplyEquations.CheckConstraintsConsistent
+                                substPatL
+                                >>= \case
+                                    (Right simplifedSubstPatL, _) ->
+                                        if substPatL == simplifedSubstPatL
+                                            then -- we are being conservative here for now and returning an error.
+                                            -- since we have already simplified the LHS, we may want to eventually return implise, but the condition
+                                            -- will contain the remainder as an equality contraint, predicating the implication on that equality being true.
 
-                                            pure . Left . RpcError.backendError . RpcError.ImplicationCheckError . RpcError.ErrorOnly . pack $
-                                                "match remainder: "
-                                                    <> renderDefault
-                                                        ( hsep $
-                                                            punctuate comma $
-                                                                map (\(t1, t2) -> pretty' @mods t1 <+> "==" <+> pretty' @mods t2) $
-                                                                    NonEmpty.toList remainder
-                                                        )
-                                        else checkImpliesMatchTerms existsL simplifedSubstPatL existsR substPatR
-                                (Left err, _) ->
-                                    pure . Left . RpcError.backendError $ RpcError.Aborted (Text.pack . constructorName $ err)
+                                                pure . Left . RpcError.backendError . RpcError.ImplicationCheckError . RpcError.ErrorOnly . pack $
+                                                    "match remainder: "
+                                                        <> renderDefault
+                                                            ( hsep $
+                                                                punctuate comma $
+                                                                    map (\(t1, t2) -> pretty' @mods t1 <+> "==" <+> pretty' @mods t2) $
+                                                                        NonEmpty.toList remainder
+                                                            )
+                                            else checkImpliesMatchTerms existsL simplifedSubstPatL existsR substPatR
+                                    (Left err, _) ->
+                                        pure . Left . RpcError.backendError $ RpcError.Aborted (Text.pack . constructorName $ err)
```

Comment:
We should probably also catch `UndefinedTerm` here... (but for now the endpoint anyway returns errors when it is unsure)